### DISCUSSION TITLE
Persistent session resending messages one by one after reconnect

### DISCUSSION THREAD

**Discussion by tommyminds** [nanomq/nanomq#1791] [Q&A]
After I reconnect a client with a persistent session, it is correctly getting each message that was sent to the topic while disconnected. It is however getting them one by one, with the interval seemingly being determined by the mqtt->retry_interval config option.

I would expect it to be possible to resend the data at a much higher interval? Or at least send messages in batches?

Is my expectation incorrect? Or am I missing a configuration option that makes this possible?

**Comment by JaylinYu** [comment]
for higher interval, just use higher retry_interval.
For aggregate batches, it is not hard to do so, but I am afraid not everybody wants batches in resend. some client just not able to deal with that amount offline msg in one time.

**Reply by tommyminds** [reply]
Thanks for the reply! I tried setting a higher interval, but I can't seem to get it below 1 second. I tried 10ms, 10Ms, 0.1s etc, but the config parser seems to not support any of that

**Reply by JaylinYu** [reply]
You didn't say you want a lower interval below 1 second...
Indeed, config parser doesn't support other metrics, and we don't recommend a frequent interval, because this value also affect the global timer, too low will result in CPU exhaustion.

**Reply by tommyminds** [reply]
Sorry for the confusion. I do indeed mean at a higher frequency, thus a lower retry_interval value, aka more than one message per second.

If we can't set this value any lower, how will a subscriber ever be able to catch up. For example, I have a producer that is sending 10 messages per second to a topic, and a consumer with a persistent session goes offline for a minute, with the current settings it will never catch up.

**Reply by JaylinYu** [reply]
I guess resending in batch would be a way out then.

**Comment by JaylinYu** [comment]
Will turn it to a feature request, planned in next release

**Reply by wanghaEMQ** [reply]
Good news. It's already in developing. https://github.com/nanomq/NanoNNG/pull/1000
